parquet-column/src/main/java/org/apache/parquet/column/statistics/BinaryStatistics.java

```diff
@@ -53,12 +53,12 @@ public void setMinMaxFromBytes(byte[] minBytes, byte[] maxBytes) {
 
   @Override
   public byte[] getMaxBytes() {
-    return max.getBytes();
+    return hasNonNullValue() ? max.getBytes() : null;
```

Comment:
should this just throw if !hasNonNullValue?
We already provide hasNonNullValue, shouldn't the caller inspect that before calling getMaxBytes?